src/org/jetbrains/plugins/scala/codeInsight/intention/IntentionUtil.scala

```diff
@@ -0,0 +1,52 @@
+package org.jetbrains.plugins.scala.codeInsight.intention
+
+import com.intellij.openapi.util.text.StringUtil
+import com.intellij.psi.impl.source.codeStyle.CodeEditUtil
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.psi.{PsiWhiteSpace, PsiComment, PsiElement}
+
+import scala.collection.mutable.ArrayBuffer
+
+object IntentionUtil {
+  def collectComments(element: PsiElement, onElementLine: Boolean = false): (Seq[PsiElement], Seq[PsiElement]) = {
+    def hasLineBreaks(whiteSpace: PsiElement): Boolean = {
+      if (!onElementLine) false
+      else StringUtil.containsLineBreak(whiteSpace.getText)
+    }
+
```

Comment:
It can be made more idiomatic with extension methods PsiElementExtTrait.prevSiblings and PsiElementExtTrait.nextSiblings.